docs/installation/docker/with-s3-storage.md

```diff
@@ -0,0 +1,104 @@
+---
+layout: default
+title: With S3 Storage
+parent: Docker
+grand_parent: Installation
+nav_order: 5
+description: "Using docker-compose to start up a Bag Database that stores files in an S3 bucket"
+permalink: /installation/docker/with-s3-storage
+---
+
+# With S3 Storage
+
+This is similar to [Without Authentication](without-authentication.md), except it uses an S3-compatible
+storage backend.  Note that doing so requires more complex configuration than is allowed through
+environment variables, so we create a custom `settings.yml` file and map that inside the Bag Database's
+container.
+
+Put this in a file named `settings.yml`:
+
+```yaml
+!com.github.swrirobotics.support.web.Configuration
+dockerHost: http://docker:2375
+driver: org.postgresql.Driver
+gpsTopics:
+    - /localization/gps
+    - gps
+    - /vehicle/gps/fix
+    - /localization/sensors/gps/novatel/raw
+    - /localization/sensors/gps/novatel/fix
+    - /imu_3dm_node/gps/fix
+    - /local_xy_origin
+jdbcPassword: letmein
+jdbcUrl: jdbc:postgresql://postgres/bag_database
+jdbcUsername: bag_database
+metadataTopics:
+    - /metadata
+vehicleNameTopics:
+    - /vms/vehicle_name
+    - /vehicle_name
+storageConfigurations:
+    - !com.github.swrirobotics.bags.storage.s3.S3BagStorageConfigImpl
+        storageId: 'YOUR_STORAGE_NAME'  # Replace with a friendly name to identify your storage backend
+        accessKey: 'YOUR_ACCESS_KEY'  # Replace with your access key
+        secretKey: 'YOUR_SECRET_KEY'  # Replace with your secret key
+        endPoint: 'YOUR_END_POINT'  # Replace with your endpoint if not using Amazon; otherwise delete this line
+        region: 'YOUR_REGION_HERE'  # Replace with your region if using Amazon; otherwise delete this line
+        bucket: 'YOUR_BUCKET_NAME'  # Replace with your bucket name
+```
+
+Then put this in `docker-compose.yml`:
+
+```yaml
+version: '3.6'
+services:
+    docker:
+        image: docker:dind
+        privileged: yes
+        networks:
+            - bagdb
+        volumes:
+            - scripts:/scripts
+            - docker_cache:/var/lib/docker
+        command: ["dockerd", "--host=tcp://0.0.0.0:2375"]
+    bagdb:
+        build: .
+        networks:
+            - bagdb
+        depends_on:
+            - postgres
+        ports:
+            - "8080:8080"
+        volumes:
+            - indexes:/root/.ros-bag-database/indexes
+            - scripts:/scripts
+            - ./settings.yml:/root/.ros-bag-database/settings.yml:ro  # Note that this is read-only so the entrypoint script won't overwrite it!
+    postgres:
+        image: postgis/postgis:11-2.5
```

Comment:
Are there potential issues with specifying the exact postgis version number? If there's ever a security update or something, are we going to need to remember to go in and change this?

Reply:
This tag isn't actually an exact version number; it's PostgreSQL 11 with PostGIS 2.5, and it should be updated with whatever patch release is available for both of them.  (this tag was updated less than a day ago, in fact)

Simply using the latest major version would potentially be a big issue.  Postgres *always* breaks compatibility between major releases; upgrading to a new major release always requires doing a manual backup and restore, so I'm hesitant to upgrade to a new major version unless there's some new feature that we really need.